### ISSUE TITLE
Heap-buffer-overflow in opj_dwt_decode_1

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#480]
Originally reported on Google Code with ID 480

```
What steps will reproduce the problem?
The original issue was raised in 
https://code.google.com/p/chromium/issues/detail?can=2&q=450844&colspec=ID%20Pri%20M%20Week%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified&id=450844.

What is the expected output? What do you see instead?
It crashes in Linux ASAN.

What version of the product are you using? On what operating system?
revision 2997. Linux ubuntu 64 bits.


Please provide any additional information below.
I attached a patch and it could fix this issue.

```

Reported by `jun_fang@foxitsoftware.com` on 2015-03-11 00:51:17

<hr>
- _Attachment: [diff_dwt.c](https://storage.googleapis.com/google-code-attachments/openjpeg/issue-480/comment-0/diff_dwt.c)_

**Comment by mayeut** [comment]
This could be fixed by commit 38770403d49528936fae44de5fc6089f0bf0c3e9 but no data to confirm

**Comment by rouault** [comment]
I looked at a few of the samples provided in https://code.google.com/p/chromium/issues/detail?can=2&q=450844&colspec=ID%20Pri%20M%20Week%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified&id=450844 and couldn't reproduce with current master. Closing
